buildrunner/validation/config.py

```diff
@@ -132,6 +138,155 @@ def validate_push(
                 if update_mp_push_tags and name is not None:
                     mp_push_tags.add(name)
 
+        def get_source_image(step: Step) -> Optional[List[str]]:
+            """
+            Get the source image from the step.build and/or step.run.image
+
+            Args:
+                step (Step): Build step
+
+            Returns:
+                Optional[str]: Source images
```

Comment:
This does not return an Optional[str], it returns a tuple. (same for the type hint above).